tools/scripts/upload-ab-to-s3.js

```diff
@@ -0,0 +1,133 @@
+const request = require('request')
+const { connectToS3, uploadToS3 } = require('./s3')
+
+var argv = require('yargs')
+  .usage('$0 [options]')
+
+  .string('env')
+  .alias('env', 'environment')
+  .describe('env', 'NR Internal Environment')
+  .default('env', 'dev')
+
+  .string('current')
+  .describe('current', 'current/stable build (defaults to -current)')
+  .default('current', 'https://js-agent.newrelic.com/nr-loader-spa-current.min.js')
+
+  .string('next')
+  .describe('next', 'next version to compare to stable version (defaults to /dev)')
+  .default('next', 'https://js-agent.newrelic.com/dev/nr-loader-spa.min.js')
+
+  .string('config')
+  .describe('config', 'NR internal config block')
+
+  .string('bucket')
+  .describe('bucket', 'S3 bucket name')
+
+  .string('role')
+  .describe('role', 'S3 role ARN')
+
+  .boolean('dry')
+  .default('dry', false)
+  .argv
+
+
+const { env, config, bucket, role, current, next, dry } = argv
+
+let counter = 0
+
+if (!env || !config || !bucket || !role) {
+  console.log("missing required param")
+  process.exit(1)
+}
+
+(async function () {
+  const filePaths = [
+    current,
+    next,
+    ...(env === 'dev' && await getOpenPrNums()).map(num => `https://js-agent.newrelic.com/pr/PR${num}`),
+  ]
+
+  Promise.all(filePaths.map(fp => getFile(fp))).then((contents) => {
+    contents = contents.filter(([url, res])=> res.statusCode === 200)
+    console.log(`found ${contents.length} valid PR builds in CDN`)
+    let output = `window.NREUM=${config};`
+    contents.forEach(([url, res, body]) => { output += wrapAgent(body) })
+    output += randomExecutor(contents.length)
+
+    const filename = `internal/${env}.js`
+
+    connectToS3(role, dry).then(async () => {
+      const uploads = await uploadToS3(filename, output, bucket, dry)
+      console.log(`Successfully uploaded ${filename} to S3`)
+      process.exit(0)
+    }).catch(err => {
+      console.log(err)
+      process.exit(1)
+    })
+  }).catch(err => {
+    console.log("error getting all files... ", err)
+  })
+
+})()
+
+function wrapAgent(agent) {
+  const agentNo = counter++
+  return `
+        function agent${agentNo}(){
+            ${agent}
+        }
+    `
+}
+
+function randomExecutor(fnCount) {
+  let output = `
+        (function (){
+            var r = Math.random();
+    `
+  for (var i = 0; i < fnCount; i++) {
```

Comment:
```suggestion
  for (var i = 1; i <= fnCount; i++) {
      output += `if (r <= ${i / fnCount}) return agent${i}();\n`
```

Reply:
Gotta remember this is generating a string template.  The return is only here to short-circuit the rest of the fn

Reply:
The output ends up looking like
```javascript
function(){
r = 0.63235
if (r <= 0.333) return agent0()
if (r <= 0.666) return agent1()
if (r <= 0.999) return agent2()
}
```

Reply:
(btw how are you getting .999 if ex. 3 / 3 => 1? I'd leave it as 1 since math.random can return infinite .9999....)